portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,6 +44,20 @@ public void init() {
   }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Create a Query instance with the kind of entity to load
+    Query query = new Query("Comment").addSort("user_comment", SortDirection.DESCENDING);
+
+    // Gather the instances of entities in Datastore of kind "Comment"
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Iterate through the entities in Datastore to get their properties
+    for (Entity entity : results.asIterable()) {  
+        String name = (String) entity.getProperty("user_name");
+        String comment = (String) entity.getProperty("user_comment");
+        messages.add("\""+ comment + "\" \n - " + name);
```

Comment:
"messages" here is a class member variable, so we would be adding the same comments to the array list every time there is an incoming GET request. Let's make this a local variable instead to avoid this issue. This way, the array object would be created every time there is a request and destroyed at the end of the function.

Reply:
Done